arctic/store/version_store.py

```diff
@@ -507,7 +507,11 @@ def append(self, symbol, data, metadata=None, prune_previous_version=True, upser
         next_ver = self._version_nums.find_one_and_update({'symbol': symbol, },
                                                           {'$inc': {'version': 1}},
                                                           upsert=False, new=True)['version']
-        if next_ver != previous_version['version'] + 1:
+
+        # Do a dirty append if previous version is the result of a restore version, or
+        # a chain of write_metadata after a restore_version.
+        # This prevents a well known issue of corrupted data.
+        if next_ver != previous_version['version'] + 1 or previous_version.get('restored_version'):
```

Comment:
Should no longer need this check?

Reply:
was left from my original implementation :-/ 
the second part of the condition is unnecessary now